course_catalog/api.py

```diff
@@ -510,3 +518,68 @@ def sync_ocw_courses(*, course_prefixes, blacklist, force_overwrite, upload_to_s
             )
         except:  # pylint: disable=bare-except
             log.exception("Error encountered parsing OCW json for %s", course_prefix)
+
+
+def sync_xpro_course_files(ids):
+    """
+    Sync all xPRO course run files for a list of course ids to database
+
+    Args:
+        ids(list of int): list of course ids to process
+    """
+    bucket = get_xpro_learning_course_bucket()
+
+    try:
+        most_recent_export = next(
+            reversed(
+                sorted(
+                    [
+                        obj
+                        for obj in bucket.objects.all()
+                        if re.search(r"/exported_courses_\d+\.tar\.gz$", obj.key)
+                    ],
+                    key=lambda obj: obj.last_modified,
+                )
+            )
+        )
+    except StopIteration:
+        log.warning("No xPRO exported courses found in xPRO S3 bucket")
+        return
+
+    course_content_type = ContentType.objects.get_for_model(Course)
+    with TemporaryDirectory() as export_tempdir, TemporaryDirectory() as tar_tempdir:
```

Comment:
Do you think devops might need to adjust anything on heroku to make sure there's enough disk space to download and extract these tar files?

Reply:
They are fairly small, in the tens of megabytes. Any comment @blarghmatey @shaidar @markbreedlove?

Reply:
Well, there's a decent amount of variability in the size of exported courses. The median course is in the tens of megabytes, but there are some outliers that reach gigabyte territory. I don't think that Heroku advertises limits to the amount of local file space that is available, so we may have to do some trials on example outlier courses to see what happens.